src/dataviews/helpers/histogram-helper.js

```diff
@@ -35,27 +29,22 @@ function trimBuckets (buckets, filledBuckets, totalBuckets) {
     : buckets;
 }
 
-helper.isShorterThan = function (limit, aggregation) {
-  var keys = _.keys(MOMENT_AGGREGATIONS);
-  var limitIndex = _.indexOf(keys, limit);
-  var aggregationIndex = _.indexOf(keys, aggregation);
-  return limitIndex > -1 && aggregationIndex > -1 && aggregationIndex < limitIndex;
-};
-
 helper.fillTimestampBuckets = function (buckets, start, aggregation, numberOfBins, from, totalBuckets) {
-  var startDate = moment.unix(start).utc();
   var filledBuckets = []; // To catch empty buckets
   var definedBucket = false;
 
   for (var i = 0; i < numberOfBins; i++) {
     definedBucket = buckets[i] !== undefined;
     filledBuckets.push(definedBucket);
 
+    var bucketStart = add(start, i, AGGREGATION_DATA[aggregation]);
```

Comment:
Would it be possible to use `buckets[i+1].start -1` for the end value instead (handling the last item)? PostgreSQL already handles a lot of stuff like changing timezones (winter/summer or [political reasons](https://www.timeanddate.com/news/time/venezuela-change-timezone.html)) so for example a day bucket could have 23.5 hours.

Reply:
Nevermind, since not all buckets are returned this is not possible. We would need to start returning start and end from windshaft first.